src/wagtail_analytics/static/wagtailanalytics/main.js

```diff
@@ -27,153 +26,58 @@ function makeCanvas(id) {
   return ctx;
 }
 
-function renderSessions(id, container) {
-  var now = moment();
-
-  var thisWeek = query({
-    ids: "ga:" + id,
-    dimensions: "ga:date,ga:nthDay",
-    metrics: "ga:sessions",
-    "start-date": moment(now).subtract(1, "day").day(0).format("YYYY-MM-DD"),
-    "end-date": moment(now).format("YYYY-MM-DD"),
-  });
-
-  var lastWeek = query({
-    ids: "ga:" + id,
-    dimensions: "ga:date,ga:nthDay",
-    metrics: "ga:sessions",
-    "start-date": moment(now)
-      .subtract(1, "day")
-      .day(0)
-      .subtract(1, "week")
-      .format("YYYY-MM-DD"),
-    "end-date": moment(now)
-      .subtract(1, "day")
-      .day(6)
-      .subtract(1, "week")
-      .format("YYYY-MM-DD"),
-  });
-
-  Promise.all([thisWeek, lastWeek]).then(function(results) {
-    var data1 = results[0].rows.map(function(row) {
-      return +row[2];
-    });
-    var data2 = results[1].rows.map(function(row) {
-      return +row[2];
-    });
-    var labels = results[1].rows.map(function(row) {
-      return +row[0];
-    });
-
-    labels = labels.map(function(label) {
-      return moment(label, "YYYYMMDD").format("ddd");
-    });
-
-    var data = {
-      labels: labels,
-      datasets: [{
+function renderSessions(container, report) {
+  this_week_integers = [];
+  last_week_integers = [];
+
+  for (var i = 0; i < 7; i++) {
```

Comment:
do not use 7 fixed here, use the length of the report array (in theory it can be less or more)